app/scripts/modules/kubernetes/src/pipeline/stages/runJob/runJobStage.js

```diff
@@ -62,10 +45,52 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
       this.stage.dnsPolicy = 'ClusterFirst';
     }
 
-    if (this.stage.container.imageDescription.fromTrigger === true) {
-      this.container = buildImageDescriptor(this.stage.container.imageDescription);
+    if (!this.stage.credentials && $scope.application.defaultCredentials.kubernetes) {
+      this.stage.credentials = $scope.application.defaultCredentials.kubernetes;
     }
 
+    if(this.stage.container) {
+      this.stage.container.imageDescription.imageId = buildImageId(this.stage.container.imageDescription);
+    }
+
+    function findContextImages(current, all, visited = {}) {
```

Comment:
I don't know if this logic should totally be shared between the run job stage and the server group dialogue..but it does feel as if you should be able to write some function that makes this a little prettier in both places. Like

```
upstreamStagesForEach(this.stage, stages, current => {
  if (current.type === 'findImage') { ... }
  if (current.type === 'bake') { ... }
})
```

Reply:
Ah, so a shared function that handles visiting each pre-req stage and implementing the result creation as a callback?

Reply:
Also, were would I need to put something like that? Is there a file where we can put shared function like this?